### ISSUE TITLE
[ADE Support]: Medallia: Intercept Survey - Pop-Up Design Concept (Accessibility Feedback Requested)

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#105190]
> [!NOTE]
> **QUICK SUMMARY:** This is the child ticket for the work being done, based on the intake ticket `104708`. The team is looking for best practices / conventional wisdom around pop-ups from an accessibility perspective. 

### Initiative

Update the design of the Medallia survey

### Desired outcome

Advise the design team so that an accessible option is selected

### Original Inquiry

> @ryan-powers-va asks: "I’m working with Chante to improve the intercept survey and one of the solutions on the table involves a pop-up window appearing with the survey after the user opts-in. What are the best practices / conventional wisdom around pop-ups from an accessibility perspective? Would love to know if they’re a no-go or allowed given specific considerations...the thought is that the opt-in modal would appear after some amount of time. If the user opts in, the survey pops up in a separate window so as to not interfere with their current task (which is a common complaint with the current survey)."

## About the Team 

- **Team**: Contact Center
- **Product name:** Medallia
- **OCTO product owner**: @clantosswett @ryan-powers-va

## Supporting Artifacts 
-  `Artifact(s) not provided, yet`

## ADE Tasks
> [!TIP]
> Customize ADE Tasks as needed. 
- [ ] **Collaborate** with @clantosswett @ryan-powers-va
- [ ] **Request** links to artifacts needing reviewed (Intercept Survey) 
- [ ] **ADE Support:** Provide best practices / conventional wisdom around pop-ups from an accessibility perspective. 
- [ ] **Document** feedback 
- [ ] **Review** with team and answer any outstanding questions

## Acceptance Criteria
> [!TIP]
> Customize ADE Acceptance Criteria as needed. 
- [ ] Feedback is received and understood by team. 
- [ ] All questions are answered. 
- [ ] Accessibility feedback is accepted and implemented.

### Ticket Updates
- [x] _Connect to an `Epic` or `Intake` (what body of work is this a part of?)_
- [x] _Label with `Originator/Team` (product team or stakeholder requesting support; ADE if internal)_
- [x] _Label date in the `Open Date` field_
- [x] _Label with `Estimate` (level of effort expected for this ticket)_
- [x] _Add `Assignee(s)` name(s) to ticket_ (include Lakwi)
- [x] _Add `Assignee(s) name(s) to each task` they will complete via handle tag (if known)_
- [x] _Select a `Priority Level`_
- [x] _Update date in `Last Checked` field_
- [ ] _Label with `Actual` (level of effort it took to complete this ticket)_
- [ ] _Update date in_ `Closed Date`
- [x] **Read** Important Notifications (ADE Internal) at the bottom of this ticket.
- [x] **Delete** Important Notifications (ADE Internal) at the bottom of this ticket, after reading.

**Comment by coforma-jamie** [comment]
Hi @ryan-powers-va, giving you feedback on your question about popups.

I'm acting under the assumption that the flow would be something like this:
1. User opts-in to popups, I'm presuming in their profile?
2. Logged-in user accesses VA.gov
3. At some point, a modal pops up, asking if they want to complete a survey
4. The user can select a link to fill out the survey, which opens in a window (not in the modal itself), OR they can close out of the modal

We're pretty concerned about this pattern. These sort of popups are highly disruptive to users - they take the user out of the flow, move focus unexpectedly. This is especially jarring for screen reader users who aren't being warned ahead of time.

This may not be a strict WCAG violation (though you can make a strong argument that it violates [Success Criterion 3.2.5 Change on Request](https://www.w3.org/TR/WCAG22/#change-on-request), and possible [Success Criterion 2.2.1 Timing Adjustable](https://www.w3.org/WAI/WCAG22/Understanding/timing-adjustable.html)).  But it **does** go against USWDS' [usability guidance for modals](https://designsystem.digital.gov/components/modal/#usability-guidance):
> **Users should trigger modals.** Modals should appear as a result of an action made by the user or (less commonly) inactivity. A modal should not surprise the user, so don’t automatically display them. Some exceptions would be to alert the user that their authenticated session in a web application is about to expire due to inactivity or if information needs to be displayed when arriving at a webpage for the first time (like accepting the use of cookies).

There's also W3C Cognitive Accessibility Guidance (called COGA), and this pattern violates the [Help Users Focus](https://www.w3.org/WAI/WCAG2/supplemental/objectives/o5-user-focus/) objective:
> Avoid using any content or elements that distract users or interrupt them. Also, consider removing content the user will find unnecessary. Provide clear headings and breadcrumbs to help users reorient and refocus if they lose focus.

**Recommendations:** We'd recommend that you _don't_ use this pattern for the above reasons. If you need to, however, we highly recommend having the popup trigger at very specific, predictable times (for example, on page load, or right before page exit), to reduce confusion. And follow USWDS' modal guidance to make sure that the modal itself is accessible.

Let me know if you have any questions, thank you!
